crates/re_data_source/src/data_loader/loader_archetype.rs

```diff
@@ -0,0 +1,157 @@
+        // TODO: If I log these times, the space views disappear and cannot be created manually
+        // if let Ok(metadata) = filepath.metadata() {
+        //     use re_log_types::{Time, Timeline};
+        //
+        //     if let Some(created) = metadata.created().ok().and_then(|t| Time::try_from(t).ok()) {
+        //         timepoint.insert(Timeline::new_temporal("created_at"), created.into());
+        //     }
+        //     if let Some(modified) = metadata
+        //         .modified()
+        //         .ok()
+        //         .and_then(|t| Time::try_from(t).ok())
+        //     {
+        //         timepoint.insert(Timeline::new_temporal("modified_at"), modified.into());
+        //     }
+        //     if let Some(accessed) = metadata
+        //         .accessed()
+        //         .ok()
+        //         .and_then(|t| Time::try_from(t).ok())
+        //     {
+        //         timepoint.insert(Timeline::new_temporal("accessed_at"), accessed.into());
+        //     }
+        // }
+
+        let extension = filepath
+            .extension()
+            .unwrap_or_default()
+            .to_ascii_lowercase()
+            .to_string_lossy()
+            .to_string();
+
+        let mut rows = Vec::new();
+
+        if crate::SUPPORTED_MESH_EXTENSIONS.contains(&extension.as_str()) {
+            re_log::debug!(?filepath, loader = self.name(), "Loading 3D model…",);
+            rows.extend(load_mesh(
+                filepath,
+                timepoint,
+                entity_path,
+                contents.into_owned(),
+            )?);
+        } else if crate::SUPPORTED_IMAGE_EXTENSIONS.contains(&extension.as_str()) {
+            re_log::debug!(?filepath, loader = self.name(), "Loading image…",);
+            rows.extend(load_image(
+                &filepath,
+                timepoint,
+                entity_path,
+                contents.into_owned(),
+            )?);
+        };
+
+        for row in rows {
+            if tx.send(row.into()).is_err() {
+                break; // The other end has decided to hang up, not our problem.
+            }
+        }
+
+        Ok(())
+    }
+}
+
+// ---
+
+fn load_mesh(
+    filepath: std::path::PathBuf,
+    timepoint: TimePoint,
+    entity_path: EntityPath,
+    contents: Vec<u8>,
+) -> Result<impl ExactSizeIterator<Item = DataRow>, DataLoaderError> {
+    re_tracing::profile_function!();
+
+    let rows = [
+        {
+            let arch = re_types::archetypes::Asset3D::from_file_contents(
+                contents,
+                re_types::components::MediaType::guess_from_path(filepath),
+            );
+            DataRow::from_archetype(RowId::new(), timepoint, entity_path.clone(), &arch)?
+        },
+        {
+            let arch = re_types::archetypes::ViewCoordinates::RIGHT_HAND_Z_UP;
+            DataRow::from_archetype(RowId::new(), TimePoint::timeless(), entity_path, &arch)?
+        },
```

Comment:
We shouldn't log `Z=up` unless we're loading a file that has that set in it (either explicitly, or implicitly based on the file extension). Otherwise, we should just fall back to whatever the default is in the viewer.